src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -62,9 +66,111 @@
+
+      if (METHOD_REF_OFFSET < 0 || SKIP_AUTOBOXING_METHOD_REF_OFFSET < 0)
+        SUPPORTS_LAMBDAS = false;
     }
   }
-  
+
+  private static int calcMethodRefOffset() {
+    LambdaTypeSample lambdaTypeSample;
+    ConstantPool constantPool;
+
+    try {
+      lambdaTypeSample = LambdaTypeSample.get();
+      constantPool = (ConstantPool) GET_CONSTANT_POOL.invoke(lambdaTypeSample.lambdaType);
+    }catch (ReflectiveOperationException e) {
+      throw new RuntimeException(e);
+    }
+
+    int constantPoolSize = constantPool.getSize();
+    for (int i = 1; i < constantPoolSize; i++) {
+      try {
+        String[] methodRefInfo = constantPool.getMemberRefInfoAt(constantPoolSize - i);
+
+        if (TypeDescriptor.getReturnType(methodRefInfo[2]).getType(
+            lambdaTypeSample.lambdaType.getClassLoader()) == lambdaTypeSample.returnType)
+          return i;
+      } catch (IllegalArgumentException e) {
+      }
+    }
+
+    return -1;
+  }
+
+  private static int calcMethodRefOffsetOfAutoboxingMethodRefLambda() {
+    AutoboxingMethodRefLambdaTypeSample lambdaTypeSample;
+    ConstantPool constantPool;
+
+    try {
+      lambdaTypeSample = AutoboxingMethodRefLambdaTypeSample.get();
+      constantPool = (ConstantPool) GET_CONSTANT_POOL.invoke(lambdaTypeSample.lambdaType);
+    }catch (ReflectiveOperationException e) {
+      throw new RuntimeException(e);
+    }
+
+    int constantPoolSize = constantPool.getSize();
+    String[] methodRefInfo = constantPool.getMemberRefInfoAt(constantPoolSize - METHOD_REF_OFFSET);
+    if (methodRefInfo[1].equals("valueOf") && constantPoolSize > 22) {
+      for (int i = METHOD_REF_OFFSET + 1; i < constantPoolSize; i++) {
+        try {
+          constantPool.getMemberRefInfoAt(constantPoolSize - i);
+          return i + 2; // '2' is MAGIC NUMBER
+        } catch (IllegalArgumentException e) {
+        }
+      }
+    }
+
+    return -1;
+  }
+
+  private static class LambdaTypeSample {
+    final Class<?> lambdaType;
+    final Class<?> returnType;
+
+    private LambdaTypeSample(Class<?> lambdaType, Class<?> returnType) {
+      this.lambdaType = lambdaType;
+      this.returnType = returnType;
+    }
+
+    static LambdaTypeSample get() throws ReflectiveOperationException {
+      Class<?> sliceOpsCls = Class.forName("java.util.stream.SliceOps");
+      Method castingArray = sliceOpsCls.getDeclaredMethod("castingArray");
+      castingArray.setAccessible(true);
+
+      return new LambdaTypeSample(castingArray.invoke(null).getClass(), Object[].class);
+    }
+  }
+
+  private static class AutoboxingMethodRefLambdaTypeSample {
+    final Class<?> lambdaType;
+
+    private AutoboxingMethodRefLambdaTypeSample(Class<?> lambdaType) {
+      this.lambdaType = lambdaType;
+    }
+
+    static AutoboxingMethodRefLambdaTypeSample get() throws ReflectiveOperationException {
+      Object collector =  Collectors.counting();
+      Field accumField = collector.getClass().getDeclaredField("accumulator");
```

Comment:
Same here - what do you think about doing this on demand the first time a user calls TypeTools?